packages/code-infra/src/utils/build.mjs

```diff
@@ -68,3 +68,63 @@ export function validatePkgJson(packageJson, options = {}) {
     throw error;
   }
 }
+
+/**
+ * Measures the performance of a function.
+ * @param {string} label - The label for the performance measurement.
+ * @param {Function} fn - The function to measure.
+ * @returns {Promise<number>} - The duration of the function execution in milliseconds.
+ */
+export async function measurePerf(label, fn) {
+  performance.mark(`${label}-start`);
+  await Promise.resolve(fn());
+  performance.mark(`${label}-end`);
+  const measurement = performance.measure(label, `${label}-start`, `${label}-end`);
+  return measurement.duration;
+}
+
+export const BASE_IGNORES = [
+  '**/*.test.js',
+  '**/*.test.ts',
+  '**/*.test.tsx',
+  '**/*.spec.js',
+  '**/*.spec.ts',
+  '**/*.spec.tsx',
+  '**/*.d.ts',
+  '**/*.test/*.*',
+  '**/test-cases/*.*',
+];
+
+/**
+ * A utility to wrap a function in a worker pool.
+ *
+ * This function will create a pool of workers and distribute the items to be processed among them.
+ * Each worker will process items sequentially, but multiple workers will run in parallel.
+ * @function
+ * @template T
+ * @param {(item: T) => Promise<void>} fn
+ * @param {Object} options
+ * @param {T[]} options.items
+ * @param {number} [options.defaultConcurrency=50]
+ * @param {'all' | 'allSettled'} [options.promiseMethod='all']
+ * @returns {Promise<void>}
+ */
+export async function wrapInWorker(fn, options) {
```

Comment:
Would like to simplify this helper a bit

```tsx
async function mapConcurrently<T, R>(items: T[], mapper: (t: T) => Promise<R>, concurrency: number): Promise<R[]>
```

* Make the signature more like a map operation, allow returning results
* Remove the `promiseMethod`, this can just be done in the `mapper` by the consumer. We can avoid leaking the implementation this way.
* I'd make `concurrency` explicitly required.